common/models/templates/test/test-tokenization.js

```diff
@@ -490,6 +491,87 @@ describe('Tokenization functions', function() {
         caretSplitsToken: true
       });
     });
+
+    it('mitigates effects of previously-distributed malformed wordbreaker output', function () {
+      const text = 'the  quick brown fox      jumped over the lazy dog  ';
+      /** @type { Context } */
+      const context = {
+        left: text,
+        right: '',
+        startOfBuffer: true,
+        endOfBuffer: true
+      }
+
+      const tokenized = models.tokenize(customWordBreakerFormer, context);
+
+      // Mitigation aims to prevent the _worst_ side-effects that can result from invalidating the
+      // underlying assumption of a monotonically-increasing index within the context -
+      // assigning repeated or blank entries the text that preceded them!
+      assert.notExists(tokenized.left.find((token) => token.text == text));
```

Comment:
To be extra clear:  without the changes to `tokenization.ts`, this assertion will fail.